File: 1/solution.go
package main

import (  
    "fmt"
    "io/ioutil"
	"strings"
)

func main() {  
    data, err := ioutil.ReadFile("input.txt")
    if err != nil {
        fmt.Println("File reading error", err)
        return
    }
	inputData := string(data)
	seaDepthArray := strings.Split(inputData, "\n")

	increasingSteps := 1
	for idx, currentSeaDepth := range seaDepthArray {
		if (idx == 0) {
			continue
		}
		prevSeaDepth := seaDepthArray[idx - 1]
		fmt.Println(idx)
		
    	if currentSeaDepth > prevSeaDepth {
			increasingSteps += 1
		}
	}

	fmt.Println("Num Increasing Steps:", increasingSteps)
}

File: 1/solution.js
const fs = require("fs");

const depthData = fs
  .readFileSync("input.txt", "utf8")
  .split("\n")
  .map((m) => parseInt(m));

const solution = depthData
  .map((currentDepth, idx) => (currentDepth > depthData[idx - 1] ? 1 : 0))
  .reduce((acc, val) => acc + val);

console.log(solution);
